dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -307,7 +305,7 @@ impl ResourceDiscovery for CommandDiscovery {
         match kind {
             DiscoveryKind::Resource => {
                 // Now we need to call discover extensions and add those resource to the list of resources
-                for extension in self.extensions.values() {
+                for extension in get_extensions().values() {
```

Comment:
I'll just note that with this change, we're intentionally trading off for memory (which is ample nowadays) and possibly some amount of speed due to the `clone()` happening on the globals, as long as I remember how Rust's clone works anyway. That seems more than reasonable to avoid the issue this PR is addressing in the first place (them not being cached).x